tests_integration/collect_test_scenarios.py

```diff
@@ -0,0 +1,93 @@
+import collections
+import os
+import random
+
+from tests_integration.lib.shared import DEFINITIONS_DIR, ENV_GITHUB_RUN_ID
+
+SKIP_TEST_CASES = {"__init__"}
+
+
+def collect_integration_test_cases() -> dict[str, list[str]]:
+    """
+    For each feedstock, return a list of all test cases that should be run for it.
+    The test cases do not include the feedstock name or the .py extension.
+
+    Example return value:
+    {
+        "feedstock1": ["aarch_migration", "version_update"],
+        "feedstock2": ["version_update"],
+    }
+
+    The return value of this function is deterministic (sorted by feedstock name and test case name).
+    """
+    test_cases = collections.defaultdict(list)
+
+    for test_case in DEFINITIONS_DIR.glob("*/*.py"):
```

Comment:
This is the part where you say it's intentionally pseudo-random, right? I would add a comment here or `random.shuffle` explicitly in a prior step.

Reply:
No, `Path.glob` is random. But we sort in the return below.

https://github.com/regro/cf-scripts/blob/cbd19ec4e914456c79dde9c2dc22a2816c719db7/tests_integration/collect_test_scenarios.py#L32-L37

The pseudorandomness is added in `get_test_scenario`.